tests/unit/test_rpm.py

```diff
@@ -82,15 +84,23 @@ def test_macros_remove():
     assert Macros.dump() == macros
 
 
+@pytest.mark.xfail(
+    rpm.__version__.split(".")[0] == "4" and int(rpm.__version__.split(".")[1]) < 17,
+    reason="Expansion and removal of built-in parametrized macros is broken before rpm 4.17",
+    raises=RPMMacroRemovalException,
+)
+def test_macros_remove_builtin_parametrized():
+    rpm.reloadConfig()
+    Macros.remove("echo")
+    assert Macros.expand("%echo abcd") == "%echo abcd"
```

Comment:
I'd parametrize this to try to remove also a macro defined with `rpm.addMacro()`. Thinking about it, it would be great to test also some rpmrc macros, both simple and parametrized, however that would require some non-trivial setup, so it's more suitable for an integration test.